cmd/appliance/upgrade/prepare.go

```diff
@@ -396,7 +404,7 @@ func prepareRun(cmd *cobra.Command, args []string, opts *prepareUpgradeOptions)
 			status := ""
 			statusChan := make(chan string)
 			defer close(statusChan)
-			go a.UpgradeStatusWorker.Watch(ctx, p, controller, appliancepkg.FileReady, appliancepkg.FileFailed, statusChan)
```

Comment:
fyi, to resolve the issue, it would have been enough to create a new context here to resolve the issue, everything else is superfluous